src/ui/public/courier/data_source/search_source.js

```diff
@@ -227,11 +226,11 @@ export function SearchSourceProvider(Promise, PromiseEmitter, Private, config) {
      * Get the parent of this SearchSource
      * @return {undefined|searchSource}
      */
-    getParent(onlyHardLinked) {
+    getParent() {
       const self = this;
       if (self._parent === false) return;
       if (self._parent) return self._parent;
-      return onlyHardLinked ? undefined : Private(RootSearchSourceProvider).get();
+      return undefined;
```

Comment:
I think we can shorten that method implementation now quite a bit to:

```
getParent() {
  return this._parent || undefined;
}
```

in which case we will only return the _parent if it's truthy and `undefined` if it's `falsey` which is exactly what above method did now.